src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -1141,6 +1119,14 @@ _mongoc_client_new_from_uri (mongoc_topology_t *topology)
 
    BSON_ASSERT (topology);
 
+   if (!topology->valid) {
+      /* This should not be reached if mongoc_client_new_from_uri_with_error and
+       * mongoc_client_pool_new_with_error raise errors when mongoc_topology_new
+       * returns an invalid topology. */
```

Comment:
This comment relates to @eramongodb's previous comment in https://github.com/mongodb/mongo-c-driver/pull/921#discussion_r774763909. Asserting `topology->valid` may be too strong, even though I don't believe this will ever be reached. I assume there's no harm in the comment -- perhaps someone can come back to this down the line for some refactoring.